appveyor.yml

```diff
@@ -1,11 +1,65 @@
 image: Visual Studio 2015
 
+environment:
+  NUGET_API_KEY:
+    secure: Rh/UHYn7HGj8a7LFWo9C1rSA0sHSQNV3wIqj9bvwRR8yLmCXu9JmbNkmkctk2cBn
+  MYGET_API_KEY:
+    secure: hA4Ut1N2lrrdEtAN24Bty/FNiU0d/Ur/dLYSqpr8jKHOvoO7MU4jD+KwzUvATh+E
+
 pull_requests:
   do_not_increment_build_number: true
 
 build_script:
 - ps: |
-    & .\build.cmd CompleteBuild
+    # A trick to test if variable is defined. Number could never be null.
```

Comment:
I don't think that this is a really good idea to move this code to the FAKE build script and reasons are following:
- This code is very AppVeyor specific. It knows a lot about environment variables and specifics of how they are filled according to different events. I don't want our FAKE script be bound to the particular CI.

- Current approach allows to split the responsibilities. The FAKE is responsible for doing the job which you ask it to do. That is the simple command (like `CompleteBuild` or `PublishNuGetPublic`) and required parameters. On other hand, AppVeyor knows about the different triggers for the build, so it's logic is to pick up the most appropriate task for that.

- It might require FAKE file redesign, as it doesn't expect that one target will _dynamically_ invoke other ones. Rather, it's supposed that you define dependencies between targets statically. If we have "CI" target we don't know its dependencies statically. Rather, it will match the current state and will need to invoke another target dynamically. I'm not sure that FAKE was even designed to handle such usage scenario and it will be possible to accomplish such a task gracefully.

- If we move all the logic to FAKE, it would be another target like "CI" with 10 parameters as input. There is no chance you are able to invoke it correctly locally, because number of parameters will be large. Also you will spend some efforts capturing the context and parsing it back in FAKE (I'm more about coding, not about performance).

- If you check the code, it's just a HUGE number of if/else statements. It will be a bit cumbersome to repeat that in FSharp and it might look less readable.

Given that I'd vote to keep it as is - CI knows about different trigger and decides which action should be applied, while FAKE is responsible for doing the actual work. Also code in AppVeyor doesn't look too complex - just a set of different if/else statments and logic to run the appropriate target.